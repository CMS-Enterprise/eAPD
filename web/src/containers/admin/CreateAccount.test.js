import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as CreateAccount,
  mapStateToProps,
  mapDispatchToProps
} from './CreateAccount';

const testProps = {
  createUser: () => {},
  error: false,
  roles: [
    { id: 1, name: "admin", isActive: true },
    { id: 2, name: "state coordinator", isActive: true },
  ],
  working: false
};

const setup = (props = {}) => {
  return shallow(<CreateAccount {...testProps} {...props} />);
};

describe('<CreateAccount />', () => {
  it('renders correctly', () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});
