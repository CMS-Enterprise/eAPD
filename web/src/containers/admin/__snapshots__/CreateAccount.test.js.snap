// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAccount /> renders correctly 1`] = `
<Fragment>
  <CardForm
    canSubmit={true}
    cancelable={true}
    error={false}
    footer={false}
    id="start-main-content"
    legend=""
    onSave={[Function]}
    primaryButtonText={
      Array [
        "Save changes",
        "Working",
      ]
    }
    sectionName="administrator"
    success={false}
    title="Create account"
    working={false}
  >
    <TextField
      ariaLabel="please enter the user's full name"
      errorPlacement="top"
      label="Name"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <TextField
      errorPlacement="top"
      label="Email"
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <Dropdown
      errorPlacement="top"
      label="State"
      name="state"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "None",
            "value": "",
          },
          Object {
            "label": "Alabama",
            "value": "al",
          },
          Object {
            "label": "Alaska",
            "value": "ak",
          },
          Object {
            "label": "Arizona",
            "value": "az",
          },
          Object {
            "label": "Arkansas",
            "value": "ar",
          },
          Object {
            "label": "California",
            "value": "ca",
          },
          Object {
            "label": "Colorado",
            "value": "co",
          },
          Object {
            "label": "Connecticut",
            "value": "ct",
          },
          Object {
            "label": "Delaware",
            "value": "de",
          },
          Object {
            "label": "Florida",
            "value": "fl",
          },
          Object {
            "label": "Georgia",
            "value": "ga",
          },
          Object {
            "label": "Hawaii",
            "value": "hi",
          },
          Object {
            "label": "Idaho",
            "value": "id",
          },
          Object {
            "label": "Iowa",
            "value": "ia",
          },
          Object {
            "label": "Illinois",
            "value": "il",
          },
          Object {
            "label": "Indiana",
            "value": "in",
          },
          Object {
            "label": "Kansas",
            "value": "ks",
          },
          Object {
            "label": "Kentucky",
            "value": "ky",
          },
          Object {
            "label": "Louisiana",
            "value": "la",
          },
          Object {
            "label": "Maine",
            "value": "me",
          },
          Object {
            "label": "Maryland",
            "value": "md",
          },
          Object {
            "label": "Massachusetts",
            "value": "ma",
          },
          Object {
            "label": "Michigan",
            "value": "mi",
          },
          Object {
            "label": "Minnesota",
            "value": "mn",
          },
          Object {
            "label": "Mississippi",
            "value": "ms",
          },
          Object {
            "label": "Missouri",
            "value": "mo",
          },
          Object {
            "label": "Montana",
            "value": "mt",
          },
          Object {
            "label": "Nebraska",
            "value": "ne",
          },
          Object {
            "label": "North Carolina",
            "value": "nc",
          },
          Object {
            "label": "North Dakota",
            "value": "nd",
          },
          Object {
            "label": "New Hampshire",
            "value": "nh",
          },
          Object {
            "label": "New Jersey",
            "value": "nj",
          },
          Object {
            "label": "New Mexico",
            "value": "nm",
          },
          Object {
            "label": "Nevada",
            "value": "nv",
          },
          Object {
            "label": "New York",
            "value": "ny",
          },
          Object {
            "label": "Ohio",
            "value": "oh",
          },
          Object {
            "label": "Oklahoma",
            "value": "ok",
          },
          Object {
            "label": "Oregon",
            "value": "or",
          },
          Object {
            "label": "Pennsylvania",
            "value": "pa",
          },
          Object {
            "label": "Rhode Island",
            "value": "ri",
          },
          Object {
            "label": "South Carolina",
            "value": "sc",
          },
          Object {
            "label": "South Dakota",
            "value": "sd",
          },
          Object {
            "label": "Tennessee",
            "value": "tn",
          },
          Object {
            "label": "Texas",
            "value": "tx",
          },
          Object {
            "label": "Utah",
            "value": "ut",
          },
          Object {
            "label": "Vermont",
            "value": "vt",
          },
          Object {
            "label": "Virginia",
            "value": "va",
          },
          Object {
            "label": "Washington",
            "value": "wa",
          },
          Object {
            "label": "West Virginia",
            "value": "wv",
          },
          Object {
            "label": "Wisconsin",
            "value": "wi",
          },
          Object {
            "label": "Wyoming",
            "value": "wy",
          },
          Object {
            "label": "District of Columbia",
            "value": "dc",
          },
          Object {
            "label": "American Samoa",
            "value": "as",
          },
          Object {
            "label": "Guam",
            "value": "gu",
          },
          Object {
            "label": "Northern Mariana Islands",
            "value": "mp",
          },
          Object {
            "label": "Puerto Rico",
            "value": "pr",
          },
          Object {
            "label": "U.S. Virgin Islands",
            "value": "vi",
          },
        ]
      }
      size="medium"
      value=""
    />
    <Dropdown
      errorPlacement="top"
      label="Authorization role"
      name="role"
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "None",
            "value": "",
          },
          Object {
            "label": "Admin",
            "value": "admin",
          },
          Object {
            "label": "State coordinator",
            "value": "state coordinator",
          },
        ]
      }
      size="medium"
      value=""
    />
    <Password
      className="mb2"
      compareTo={
        Array [
          "",
          "",
        ]
      }
      customErrorMessage=""
      errorMessage=""
      onChange={[Function]}
      showMeter={true}
      title="Password"
      value=""
    />
  </CardForm>
</Fragment>
`;
